client/src/redux/donation-saga.js

```diff
@@ -73,10 +74,45 @@ function* postChargeStripeSaga({ payload }) {
   }
 }
 
-function* postChargeStripeCardSaga({ payload }) {
+function* stripeCardErrorHandler(
+  error,
+  handleAuthentication,
+  clientSecret,
+  paymentMethodId
+) {
+  if (error.type === 'UserActionRequired' && clientSecret) {
+    yield handleAuthentication(clientSecret, paymentMethodId)
+      .then(result => {
+        if (
+          !result.paymentIntent?.status ||
+          !result.paymentIntent.status === 'succeeded'
+        )
```

Comment:
```suggestion
        if (result?.paymentIntent?.status !== 'succeeded')
```
It's supposed to throw unless the status is 'succeeded', no?